gen/__init__.py

```diff
@@ -544,7 +537,28 @@ def generate(
                 late_variables[setter.name] = setter.late_expression
     log.debug('Late variables:\n{}'.format(pprint.pformat(late_variables)))
 
-    argument_dict = {k: v.value for k, v in resolver.arguments.items() if v.is_finalized}
+    return late_variables
+
+
+def get_final_arguments(resolver):
+    return {k: v.value for k, v in resolver.arguments.items() if v.is_finalized}
+
+
+def generate(
+        arguments,
+        extra_templates=list(),
+        extra_sources=list(),
+        extra_targets=list()):
+    # To maintain the old API where we passed arguments rather than the new name.
+    user_arguments = arguments
+    arguments = None
```

Comment:
This is copy/paste code and I do not want to be distracted by inherited issues. If you feel this is worth addressing, we can open a PR and/or JIRA

That being said, the point of this seems to be so that you can call gen.generate(arguments={}) as opposed to the more correct gen.generate(user_arguments={}). The distinction being that the name arguments implies the general complete set, whereas user arguments are a non-optional source.